### PR TITLE
Add in sqlite3 to permit rails installation

### PR DESCRIPTION
**Pull request by luke-hill**
I'm not 100% this will fix everything. But I couldn't find a way of testing "in-container" as trying to run `sudo` commands is banned

Possibly as another option, could we permit people to manually edit stuff inside the container. That would allow us to be certain of fixes pushed here beforehand.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -196,6 +196,12 @@ RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key
         libxss1 \
     && rm -rf /var/lib/apt/lists/*
 
+# Install sqlite3 - Required for cucumber-rails
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
```
Comment (aurelien-reeves) [inline Dockerfile]: To be consistent, could you please use the long version `--assume-yes` instead of `-y`?

**Comment by mpkorstanje** [comment]
You can run run `docker build` locally and then use the image created that way to install rails.

**Review by aurelien-reeves — CHANGES_REQUESTED** [review]
Dependencies seems still missing, I guess the `-dev` ones and what is needed for compilation.

```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /var/lib/gems/2.7.0/gems/sqlite3-1.4.2/ext/sqlite3
/usr/bin/ruby2.7 -I /usr/lib/ruby/2.7.0 -r ./siteconf20210326-418-4rdjxs.rb extconf.rb
checking for sqlite3.h... no
sqlite3.h is missing. Try 'brew install sqlite3',
'yum install sqlite-devel' or 'apt-get install libsqlite3-dev'
and check your shared library search path (the
location where your sqlite3 shared library is located).
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.
```

To be able to test with cucumber-rails, from within the cucumber-rails repo:

    docker run --publish 6006:6006 --volume "$PWD":/app --volume "$HOME/.m2/repository":/home/cukebot/.m2/repository --user 0 --rm -it cucumber/cucumber-build:latest bash

`--user 0` will bring you into the container as root so, this is just for testing purpose. That way you'll be able to `apt install` things and make your tests

Note: $PWD works with zsh, don't know if it works the same with bash.

File: Dockerfile (hunk 1 above)
Comment (luke-hill) [reply Dockerfile]: This was what is in the repo. I copied that?

Comment (aurelien-reeves) [reply Dockerfile]: Indeed, a bunch of `-y` remains 😅

**Comment by aurelien-reeves** [comment]
With your changes, I've been able to `bundle install` inside the container.

But `bin/install_geckodriver.sh` and `bin/install_webpacker.sh` are failing because they rely on `sudo`.

Do you plan something next? What the definitive purpose of that PR regarding cucumber-rails? Being able to execute all the tests in the container?

**Review by aurelien-reeves — APPROVED** [review]

**Comment by luke-hill** [comment]
I've got future PR's planned for those.
install_geckodriver will be deleted. And will be delegating to a gem that exists and does the work 10x better.
install_webpacker. Not sure yet. Can we come back to that later?

**Comment by mpkorstanje** [comment]
I thought just occurred to me. It would be prudent to limit the scope of the `cucumber-build` image to the monorepo. Otherwise it will be very hard to validate if changes we make would potentially break builds elsewhere -- and also which dependencies are used and which are unused.

So will be this be used in the mono-repo?

**Comment by luke-hill** [comment]
Probably will be yes. But this is why the work from aslak on versioning of this is also prudent. I think versioning or cutting this Dockerfile being registered is the better way to go.

**Pull request merged**